HacktoberfestProject.Web/Models/Entities/UserEntity.cs

```diff
@@ -1,51 +1,52 @@
-﻿using Microsoft.Azure.Cosmos.Table;
-using Microsoft.Azure.Documents;
-using Newtonsoft.Json;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
-
-namespace HacktoberfestProject.Web.Models.Entities
-{
-    public class UserEntity : TableEntity
-    {
-        public Guid UserId { get; set; }
-
-        [IgnoreProperty]
-        public List<RepositoryEntity> RepositoryPrAddedTo { get; set; }
-
-        public UserEntity()
-		{
-
-		}
-
-        public UserEntity(Guid id, List<RepositoryEntity> repositoryPrAddedTo = null)
-        {
-            UserId = id;
-            RepositoryPrAddedTo = repositoryPrAddedTo;
-            PartitionKey = "Users";
-            RowKey = UserId.ToString();
-            
-        }
-
-        public override IDictionary<string, EntityProperty> WriteEntity(OperationContext operationContext)
-        {
-            // This line will write partition key and row key, but not properties with IgnoreProperty attribute
-            var propertiesToBeWritten = base.WriteEntity(operationContext);
-
-            // Writing x manually as a serialized string.
-            propertiesToBeWritten["jsonDetails"] = new EntityProperty(JsonConvert.SerializeObject(this.RepositoryPrAddedTo));
-            return propertiesToBeWritten;
-        }
-
-        public override void ReadEntity(IDictionary<string, EntityProperty> properties, OperationContext operationContext)
-        {
-            base.ReadEntity(properties, operationContext);
-            if (properties.ContainsKey(nameof(this.RepositoryPrAddedTo)))
-            {
-                this.RepositoryPrAddedTo = JsonConvert.DeserializeObject<List<RepositoryEntity>>(properties["jsonDetails"].StringValue);
-            }
-        }
-    }
-}
+﻿using System.Collections.Generic;
+
+using Microsoft.Azure.Cosmos.Table;
+using Newtonsoft.Json;
+
+namespace HacktoberfestProject.Web.Models.Entities
+{
+    public class UserEntity : TableEntity
+    {
+        public const string PARTITIONKEY = "Users";
+
+        [IgnoreProperty]
+        public string Username
+        {
+            get => RowKey;
+            set => RowKey = value;
+        }
+
+        [IgnoreProperty]
+        public List<RepositoryEntity> RepositoryPrAddedTo { get; set; }
+
+        public UserEntity()
+		{
+            PartitionKey = PARTITIONKEY;

```

Comment:
I think their needed to be an empty default constructor in the TableContext. @CrypticEngima can you confirm please?

Reply:
There needs to be a parameter-less one, empty is not explicitly necessary.